lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6570,11 +6571,13 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       return;
 
     // Require the carried type to be materializable.
-    auto IFacTy = EED->getArgumentInterfaceType();
-    if (IFacTy && !IFacTy->isMaterializable()) {
-      TC.diagnose(EED->getLoc(), diag::enum_element_not_materializable, IFacTy);
-      EED->setInterfaceType(ErrorType::get(TC.Context));
-      EED->setInvalid();
+    if (auto argTy = EED->getArgumentInterfaceType()) {
+      auto argTupleTy = argTy->getAs<TupleType>();
+      if ((argTupleTy && argTupleTy->hasInOutElement()) || argTy->isLValueType()) {
```

Comment:
I don't think argTy->isLValueType() can happen here; you should assert against that instead